source/declarations.tex

```diff
@@ -4449,6 +4469,326 @@
+int g() post(b1);       // error: mismatched \grammarterm{result-name-introducer} presence
+
+namespace N {
+  void h() pre (b1);
+  bool b1;
+  void h() pre (b1);    // error: \grammarterm{function-contract-specifier}s differ according to
+                        // the one-definition rule\iref{basic.def.odr}.
+}
+\end{codeblock}
+\end{example}
+
+\pnum
+A virtual function\iref{class.virtual},
+a deleted function\iref{dcl.fct.def.delete},
+or a function defaulted on its first declaration\iref{dcl.fct.def.default}
+shall not have a \grammarterm{function-contract-specifier-seq}.
+
+\pnum
+If the predicate of a postcondition assertion
+of a function \placeholder{f}
+odr-uses\iref{basic.def.odr}
+a non-reference parameter of \placeholder{f},
+that parameter
+and the corresponding parameter on all declarations of \placeholder{f}
+shall have \keyword{const} type.
+\begin{note}
+This requirement applies
+even to declarations that do not specify the \grammarterm{postcondition-specifier}.
+Parameters with array or function type
+will decay to non-\keyword{const} types
+even if a \keyword{const} qualifier is present.
+\begin{example}
+\begin{codeblock}
+int f(const int i[10])
+  post(r : r == i[0]);  // error: \tcode{i} has type \tcode{const int *} (not \tcode{int* const}).
+\end{codeblock}
+\end{example}
+\end{note}
+
+\pnum
+\begin{note}
+The function contract assertions of a function
+are evaluated even when invoked indirectly,
+such as through a pointer to function or a pointer to member function.
+A pointer to function,
+pointer to member function,
+or function type alias
+cannot have a \grammarterm{function-contract-specifier-seq}
+associated directly with it.
+\end{note}
+
+\pnum
+The function contract assertions of a function
+are considered to be \defnx{needed}{needed!function contract assertion}\iref{temp.inst} when
+\begin{itemize}
+\item
+the function is odr-used\iref{basic.def.odr} or
+\item
+the function is defined.
+\end{itemize}
+\begin{note}
+Overload resolution does not consider
+\grammarterm{function-contract-specifier}s\iref{temp.deduct,temp.inst}.
+\begin{example}
+\begin{codeblock}
+template <typename T>  void f(T t) pre( t == "" );
+template <typename T>  void f(T&& t);
+void g()
+{
+  f(5);     // error: ambiguous
+}
+\end{codeblock}
+\end{example}
+\end{note}
+
+
+\rSec2[dcl.contract.res]{Referring to the result object}
+
+\begin{bnf}
+\nontermdef{attributed-identifier}\br
+  identifier \opt{attribute-specifier-seq}
+\end{bnf}
+
+\begin{bnf}
+\nontermdef{result-name-introducer}\br
+  attributed-identifier \terminal{:}
+\end{bnf}
+
+\pnum
+The \grammarterm{result-name-introducer}
+of a \grammarterm{postcondition-specifier}
+is a declaration.
+The \grammarterm{result-name-introducer}
+introduces the \grammarterm{identifier}
+as the name of a \defn{result binding}
+of the associated function.
+If a postcondition assertion has a \grammarterm{result-name-introducer}
+and the return type of the function is $cv$ \keyword{void},
```

Comment:
Please use `\cv{}` instead of `$cv$` (global search-replace).

Reply:
done